src/ca/mcgill/ecse211/lab4/LightLocalizer.java

```diff
@@ -0,0 +1,255 @@
+
+
+/**
+ * This clas implements light localization to complete the localization process for the robot. It
+ * detects a given amount of black lines on the test surface. Then, it uses the angles at which the
+ * robot detected those lines to orient itself towards the origin.
+ * 
+ * @author1 Cristian Ciungu
+ * @author2 Hao Shu
+ * @version 12-02-2019
+ * 
+ * 
+ */
+package ca.mcgill.ecse211.lab4;
+
+import lejos.hardware.Sound;
+import lejos.hardware.ev3.LocalEV3;
+import lejos.hardware.motor.EV3LargeRegulatedMotor;
+import lejos.hardware.sensor.EV3ColorSensor;
+import lejos.robotics.SampleProvider;
+import ca.mcgill.ecse211.odometer.*;
+import java.util.Arrays;
+
+public class LightLocalizer implements Runnable {
+
+  // robot constants
+  public final static int ROTATION_SPEED = UltrasonicLocalizer.ROTATION_SPEED;// the speed for the
+                                                                              // rotation of the
+                                                                              // robot
+  private final static double SENSOR_LENGTH = 15.00;// the length from the head of the robot to the
+                                                    // Sensor
+
+  private final static int FORWARD_SPEED = 80;// the speed for the robot to go forward
+
+  private Odometer odometer;
+  private EV3LargeRegulatedMotor leftMotor, rightMotor;
+  private static final EV3ColorSensor ColorSensor =
+      new EV3ColorSensor(LocalEV3.get().getPort("S2"));// Instantiate the EV3 ColorSensor
+  private double light_received;
+
+  private SampleProvider Color;
+  float[] colorData;
+  int counter = 0; // a counter is used to record the number of black lines detected
+  double[] line_num;// an array to record the angle of the robot at each black line

```

Comment:
In Java, `camelCase` conventions are used, unlike Python, where words in variable names are `separated_by_underscores`.